aws/restore/restore.go

```
@@ -0,0 +1,277 @@
+package restore
+
+import (
+	"encoding/json"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/herdius/herdius-core/blockchain"
+	"github.com/herdius/herdius-core/blockchain/protobuf"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/session"
+	"github.com/aws/aws-sdk-go/service/s3"
+
+	"github.com/herdius/herdius-core/config"
+)
+
+type RestorerI interface {
+	Restore() error
+	testCompleteChainRemote() (bool, error)
+	clearOld() error
+	downloadChain() (*[]protobuf.BaseBlock, error)
+	replayChain(*[]protobuf.BaseBlock) error
+}
+
+type Restorer struct {
+	statePath       string
+	chainPath       string
+	s3bucket        string
+	heightToRestore int
+	s3              *s3.S3
+}
+
+func NewRestorer(env string, height int) RestorerI {
+	detail := config.GetConfiguration(env)
+	s := s3.New(session.New())
+	return Restorer{
+		statePath:       detail.StateDBPath,
+		chainPath:       detail.ChainDBPath,
+		s3bucket:        detail.S3Bucket,
+		heightToRestore: height,
+		s3:              s,
+	}
+}
+
+// Restore retrieves and procceses an entire blockchain stored in S3
+// into the Supervisor's local blockchain and statedb
+func (r Restorer) Restore() error {
+	succ, err := r.testCompleteChainRemote()
+	if err != nil {
+		err = fmt.Errorf("restore failed while trying to test remote chain: %v", err)
```

this error is never handled, shadow by the one in https://github.com/herdius/herdius-core/pull/76/files#diff-22e5ba109d64d8c70583a7f1b780efd3R59